File: easy/ts/algoExpert-transpose-matrix.ts
export function transposeMatrix(matrix: number[][]) {
  let answerArr: number[][] = [];
  let i = 0;
  let j = 0;
  while (i < matrix[0].length) {
    let elementArr: number[] = [];
    while (j < matrix.length) {
      elementArr.push(matrix[j][i]);
      j++;
    }
    answerArr.push(elementArr);
    i++;
    j = 0;
  }
  return answerArr;
}

File: easy/python/algo_expert_transpose_matrix.py
def transposeMatrix(matrix):
    answer = []
    i = 0
    j = 0
    while i < len(matrix[0]):
        element_arr = []
        while j < len(matrix):
            element_arr.append(matrix[j][i])
            j += 1
        answer.append(element_arr)
        j = 0
        i += 1
    return answer
